### DISCUSSION TITLE
Are there plans to open outgoing ports to support custom storage backends

### DISCUSSION THREAD

**Discussion by lutengda** [grafana/tempo#3839] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by mapno** [comment]
Hi @lutengda, can you tell us more? We support S3, GCS and Azure compatible backends. You'd like support for one not in the list?

**Reply by lutengda** [reply]
Yes, I would like to support a traditional relational database or a time series database as the storage backend, which would provide more flexibility in the choice of storage backend and provide better scalability and high availability

**Reply by lutengda** [reply]
I think tempo is a great monitoring solution, and more people will choose to use it if they can support custom storage backends

**Reply by mapno** [reply, marked as answer]
Tempo is built around object storage and uses a column-oriented format (Apache Parquet) to store and read traces. How do you see a relational database improving or extending Tempo? That's a massive change on how Tempo operates.

> or a time series database

Tempo stores traces, not time series. I'm not sure I understand what you mean by that.

> if they can support custom storage backends

Also, I don't think I understand what you mean by custom storage backend.

What I'd see more realistic is adding support for a object storage solution that's not S3, GCS or Azure blob storage. Is there one you'd like to see support for?

**Reply by lutengda** [reply]
Ok, I see, maybe my idea is not too mature, thank you for your answer
